examples/step-5/step-5.cc

```diff
@@ -113,6 +113,14 @@ double coefficient (const Point<dim> &p)
   else
     return 1;
 }
+// @note It looks like we use a discontinuous coefficient, however,
+// the finite element method using quadrature really sees only a
+// polynomial approximation (defined by a
+// <code>quadrature_formula</code>) of that discontinuous coefficient
+// on every mesh. (Though that polynomial approximation of course
+// tends to a discontinuous function under mesh refinement.) Refer to
+// CellAccessor::set_material_id to set a discontinuous coefficient
```

Comment:
> Consider a quadrature point located on the boundary of two cells with material1 and material2

We typically use Gauss quadratures, so no quadrature point will be directly on the boundary between cells. You are of course correct, that you could pick a different quadrature and change the coefficient function based on the current cell, but I agree with JP that this probably shouldn't be mentioned at this place.